official/utils/flags/core.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import absl.flags
+
+from official.utils.flags import _base
+from official.utils.flags import _performance
+
+
+def define_base(*args, **kwargs):
+  key_flags = _base.define_base(*args, **kwargs)
+  [absl.flags.declare_key_flag(fl) for fl in key_flags]
+
```

Comment:
Why do we have this extra level of define? Can't we just mark the flags to be key flag when they are declared?

Reply:
If declared when `flags.DEFINE_*` is called then they are key flags only for that module. However the declaration needs to be in `core.py` in order for the `adopt_module_key_flags()` chaining to work correctly. `DEFINE`s are magical context-aware little beasts.